app/src/main/java/com/example/mytestproject/ui/FragmentSecond.kt

```diff
@@ -17,4 +18,14 @@ class FragmentSecond : Fragment() {
         return inflater.inflate(R.layout.fragment_second, container, false)
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
```

Comment:
We need to do it for every new fragment that we add to app. It is easy to forget to do it.
Probably, it is better to show menu only in weather fragments.  

Also there is an option - to create a separate WeatherFragment that will be a container for three nested weather fragments (today, 3 and 10). It is a bit more complicated, but menu will be shown only once - in this container fragment.

Reply:
Fixed